gigascience/gigadb-website
# New feature/2066 edit attr modal

# Pull request for issue: #2066

IMPORTANT: this is a follow up of https://github.com/gigascience/gigadb-website/pull/2082, please merge that PR before this one

Wrap the edit attr form in a modal (see comments in https://github.com/gigascience/gigadb-website/pull/2082#pullrequestreview-2511059933) so that the layout and UX is a bit less cramped and (hopefully) more straightforward, and to easily solve the layout issue expressed in the conversation

## How to test?

- navigate to http://gigadb.gigasciencejournal.com/adminFile/update/id/13973
- click on edit button
- interact with modal:
  - keyboard tabbing: expect focus trap on modal
  - cancel modal: return to previous state unchanged
  - click "save attribute": expect to see the updated attribute

## How have functionalities been implemented?

Wrapped the edit form in a modal that gets rendered with the reponse from the `actionEditAttr` method of the controller, everything else remains unchanged.

A test was removed since it tested an edge case which is no longer possible. The test checked that a fringe interaction (new attribute and edit attribute being used simultaneously) did not lead to errors. I think the fact that we can remove that test is a good indication at this edge case is no longer possible, as the new layout restricts the user to handle the editted attribute before doing anything else

## Any changes to automated tests?

One test was deleted, other tests stay the same as the actions to edit attributes stay the same
